### ISSUE TITLE
Multiple roslyn analyzers cannot be created by MSBuild

### ISSUE THREAD

**Issue by triforcely** [SonarSource/sonar-csharp#1570]
### Description

When running VSTS build with SonarQube following warning are displayed and finished build reports no issues with code:

```
CSC (0, 0)
CSC(0,0): Warning CS8032: An instance of analyzer RoslynSecurityGuard.Analyzers.XxeAnalyzer cannot be created from C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube\resources\1\analyzers\dotnet\cs\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..
CSC (0, 0)
CSC(0,0): Warning CS8032: An instance of analyzer RoslynSecurityGuard.Analyzers.Taint.TaintAnalyzer cannot be created from C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube\resources\1\analyzers\dotnet\cs\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..
CSC (0, 0)
CSC(0,0): Warning CS8032: An instance of analyzer RoslynSecurityGuard.Analyzers.XxeAnalyzer cannot be created from C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube\resources\1\analyzers\dotnet\cs\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..
CSC (0, 0)
CSC(0,0): Warning CS8032: An instance of analyzer RoslynSecurityGuard.Analyzers.Taint.TaintAnalyzer cannot be created from C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube\resources\1\analyzers\dotnet\cs\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..

```

### Repro steps

In my case, just add SonarQube steps to existing build for .NET Framework solution in VSTS.

### Expected behavior

Roslyn reports something.

### Actual behavior

*RoslynCA.json files contain:

```
{
  "$schema": "http://json.schemastore.org/sarif-1.0.0",
  "version": "1.0.0",
  "runs": [
    {
      "tool": {
        "name": "Microsoft (R) Visual C# Compiler",
        "version": "2.8.0.0",
        "fileVersion": "2.8.3.62923 (7aafab56)",
        "semanticVersion": "2.8.0",
        "language": "en-US"
      },
      "results": [
        {
          "ruleId": "CS8032",
          "level": "warning",
          "message": "An instance of analyzer RoslynSecurityGuard.Analyzers.XxeAnalyzer cannot be created from C:\\Users\\tfsbuild\\AppData\\Local\\Temp\\.sonarqube\\resources\\1\\analyzers\\dotnet\\cs\\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..",
          "properties": {
            "warningLevel": 1
          }
        },
        {
          "ruleId": "CS8032",
          "level": "warning",
          "message": "An instance of analyzer RoslynSecurityGuard.Analyzers.Taint.TaintAnalyzer cannot be created from C:\\Users\\tfsbuild\\AppData\\Local\\Temp\\.sonarqube\\resources\\1\\analyzers\\dotnet\\cs\\RoslynSecurityGuard.dll : Exception has been thrown by the target of an invocation..",
          "properties": {
            "warningLevel": 1
          }
        }
      ],
      "rules": {
        "CS8032": {
          "id": "CS8032",
          "shortDescription": "An analyzer instance cannot be created",
          "defaultLevel": "warning",
          "properties": {
            "category": "Compiler",
            "isEnabledByDefault": true,
            "tags": [
              "Compiler",
              "Telemetry"
            ]
          }
        }
      }
    }
  ]
}
```

### Related information

* SonarC# Version: VSTS "Prepare Analysis Configuration" and "Run Code Analysis", v4.0
* Visual Studio Version: MSBuild 15.0
* VSTS Agent: Self-hosted agent, maybe I'm missing some prerequisites?

**Comment by triforcely** [comment]
Alright, I think I found the culprit. "Prepare Analysis Configuration" task is supposed to install roslyn analyzers and in my case it installs it in "C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube". I investigated contents os this file and found out that zip files containing assemblies were not fully extracted. When running next builds, this step did not re-extract files and analyzers weren't able to find it's dependencies. I fixed it by removing "C:\Users\tfsbuild\AppData\Local\Temp\.sonarqube" folder and agent re-installed them fine. I think that build tasks should assert if files were extracted fully to save some headaches in the future.
